File: front/src/services/topic.ts
import { IFavorite } from '../models/IFavorite'
import { ITopic } from '../models/ITopic'
import { query } from './query'

export function getTopics(key: string): Promise<ITopic[]> {
  if (!key) return Promise.resolve([])
  return query('GetTopics', key)
}

export function getFavoriteTopics(): Promise<IFavorite[]> {
  return query('GetFavoriteTopics')
}

export function toggleTopicFavorite(favorite: IFavorite): Promise<void> {
  return query('ToggleFavoriteTopic', favorite)
}

File: back/src/controllers/topic.py
from config import config


def getTopics(key: str):
    return []


def getFavoriteTopics():
    return list(
        map(
            lambda favorite: {
                "server": config.servers[favorite.server],
                "name": favorite.topic,
            },
            config.favorites,
        )
    )


def toggleTopicFavorite(key: str, topic: str):
    return None
